### ISSUE TITLE
Questions about evaluations

### ISSUE THREAD

**Issue by haruishi43** [yun-liu/DDS#1]
After reading your paper, I noticed that you have used Gated-SCNN for your comparison.
As far as I know, GSCNN outputs semantic segmentation and category agnostic edge maps.
In the paper, you described the process of evaluation as "Gated-SCNN learns semantic edges for improving the training of semantic segmentation. Hence, we retain it for semantic edge detection by removing its segmentation loss and dual task loss, and the other settings are kept by default."
I'm curious on how to extract semantic edges in Gated-SCNN since looking at their code and paper, it seems like the edges are binary maps (maybe you could mask the segmentation maps with the edge maps, but it doesn't seem like what you guys did).
Could you explain the process?

**Comment by yun-liu** [comment]
To be honest, it is not a good idea to compare with GSCNN, but one of our reviewers asked us to do so. One on hand, GSCNN is for semantic segmentation, not for semantic edge detection. If we directly mask the segmentation maps with the binary edge maps as you said, the performance for semantic edge detection would be poor. On the other hand, GSCNN is not implemented in the same way as its paper. For example, please see the following code for computing its [dual task loss](https://github.com/nv-tlabs/GSCNN/blob/master/my_functionals/DualTaskLoss.py):
https://github.com/nv-tlabs/GSCNN/blob/master/my_functionals/custom_functional.py#L93
https://github.com/nv-tlabs/GSCNN/blob/master/my_functionals/custom_functional.py#L39
The code **does not execute** due to the usage of early `return`.

Therefore, we decide to remove their loss functions and impose the same loss functions as our DDS to it. Specifically, we impose the semantic edge loss function (with thick semantic edges as the ground-truth) in DDS to supervise its semantic segmentation prediction, so GSCNN would predict the semantic edge maps. Besides, we impose the binary edge loss function (with thin binary edges as the ground-truth) in DDS to supervise its binary edge maps, so the binary edge maps are still for auxiliary supervision. However, it is still **unfair** because GSCNN uses the much larger **WideResNet** while our DDS uses the standard **ResNet**.

You can see that this experiment is not worth paying much attention to. Please forget it! If you need the results, please directly use the results in our paper.

**Comment by haruishi43** [comment]
@yun-liu Thanks for the response! I was trying to find comparison methods for semantic edge detection and came across your paper. I was also trying to compare GSCNN since it was being compared in HED-UNet as well and was wondering how you guys got it to train. Thanks for the heads up.
I admire the work you put in into evaluating your methods!
